### PR TITLE
Enable CI tests and code linting

### PR DESCRIPTION
**Pull request by benoitdm-oslandia**
* Fix github CI to run tests
* add pre-commit tool to improve code quality (for future commits), a dedicated PR to fix all the typo could be created later

### CODE REVIEW

File: internal/cql/cql_parser.go [hunk 1]
```diff
@@ -539,8 +539,6 @@ func (p *CQLParser) booleanValueExpression(_p int) (localctx IBooleanValueExpres
 	var _parentctx antlr.ParserRuleContext = p.GetParserRuleContext()
 	_parentState := p.GetState()
 	localctx = NewBooleanValueExpressionContext(p, p.GetParserRuleContext(), _parentState)
-	var _prevctx IBooleanValueExpressionContext = localctx
```
Comment (dr-jts) [inline internal/cql/cql_parser.go:542]: This file is auto-generated, so it's problematic to change it.  Can it be removed from linting?

Comment (benoitdm-oslandia) [reply internal/cql/cql_parser.go]: Done!

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# CHECK for new hook to add (JS, etc.): https://pre-commit.com/hooks.html
```
Comment (pramsey) [inline .pre-commit-config.yaml:1]: Can you back this out into a separate PR? I don't like adding build deps as a "surprise!"

Comment (benoitdm-oslandia) [reply .pre-commit-config.yaml:1]: Sorry, I do not understand what building deps it adds. Can you elaborate please?

Comment (dr-jts) [reply .pre-commit-config.yaml:1]: Are the pre-commit checks purely a developer client-side check?   So they have no effect on the CI build chain (or external ones)?

Comment (benoitdm-oslandia) [reply .pre-commit-config.yaml:1]: Exactly, these are purely dev side checks and are not mandatory to work. They may affect the CI if you integrate the pre-commit checks in the CI (which is not the case).

Comment (Guts) [reply .pre-commit-config.yaml:1]: Hi,

As a follower of this work and an intensive user of git hooks through pre-commit, I just want to mention that it is possible to plug https://pre-commit.ci/ in order to run it automatically on PR.

Comment (benoitdm-oslandia) [reply .pre-commit-config.yaml:1]: Thanks @Guts I think we will soon add theses checks in the github CI

**Pull request merged**